contracts/src/periphery/cert/router/EigenDACertVerifierRouter.sol

```diff
@@ -0,0 +1,88 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.9;
+
+import {IEigenDACertVerifierBase, IEigenDACertVerifier} from "src/periphery/cert/interfaces/IEigenDACertVerifier.sol";
+import {IEigenDACertVerifierRouter} from "src/periphery/cert/interfaces/IEigenDACertVerifierRouter.sol";
+import {OwnableUpgradeable} from "lib/openzeppelin-contracts-upgradeable/contracts/access/OwnableUpgradeable.sol";
+
+contract EigenDACertVerifierRouter is IEigenDACertVerifierRouter, OwnableUpgradeable {
+    /// @notice A mapping from an activation block number (ABN) to a cert verifier address.
+    mapping(uint32 => address) public certVerifiers;
+
+    /// @notice The list of Activation Block Numbers (ABNs) for the cert verifiers.
+    /// @dev The list is guaranteed to be in ascending order
+    ///      and corresponds to the keys of the certVerifiers mapping.
+    uint32[] public certVerifierABNs;
+
+    event CertVerifierAdded(uint32 indexed activationBlockNumber, address indexed certVerifier);
+
+    error ABNNotInFuture(uint32 activationBlockNumber);
+    error ABNNotGreaterThanLast(uint32 activationBlockNumber);
+    error InvalidCertLength();
+
+    /// IEigenDACertVerifierRouter ///
+
+    /// @inheritdoc IEigenDACertVerifierBase
+    function checkDACert(bytes calldata abiEncodedCert) external view returns (uint8) {
+        return IEigenDACertVerifier(getCertVerifierAt(_getRBN(abiEncodedCert))).checkDACert(abiEncodedCert);
+    }
+
+    function getCertVerifierAt(uint32 referenceBlockNumber) public view returns (address) {
+        return certVerifiers[_findPrecedingRegisteredABN(referenceBlockNumber)];
+    }
+
+    /// ADMIN ///
+
+    function initialize(address _initialOwner, address certVerifier) external initializer {
+        _transferOwnership(_initialOwner);
+        // Add a default cert verifier at block 0, which will be used for all blocks before the first ABN.
+        _addCertVerifier(0, certVerifier);
+    }
+
+    function addCertVerifier(uint32 activationBlockNumber, address certVerifier) external onlyOwner {
```

Comment:
Maybe it's already been brought up-

would it be possible to assert that the input address implements the expected interface?

Reply:
There's https://0xjac.github.io/EIPs/EIPS/eip-165 that exists but it looks a bit clumsy and complex. Patrick would know better whether its worth the effort though

Reply:
It is not generally possible. I am very against adding EIP-165 support since IMO it is a technique that adds boilerplate without adding any real safety guarantees. There is currently no such assertion, in a similar vein to how you cannot really verify the correctness of an FFI to a general binary.